### PR TITLE
Allow to mount extra volumes in otel-collector(-metrics) pods

### PR DESCRIPTION
**Pull request by rchallie**
Hi !

Allow to mount extra volumes permit, for exemple, to mount a volume with targets files to use prometheus `file_sd_config` integration.

> No indentation is not an artistic choice, it's because for `volumes` the first line with the `-` is to much indented and for `volumeMounts` an empty line appear before the first element of `extraVolumeMounts`.

Thanks ! Have a good day !

### CODE REVIEW

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml [hunk 1]
```diff
@@ -111,6 +111,9 @@ spec:
           volumeMounts:
             - name: otel-collector-metrics-config-vol
               mountPath: /conf
+{{- if .Values.otelCollectorMetrics.extraVolumeMounts }}
+{{- toYaml .Values.otelCollectorMetrics.extraVolumeMounts | nindent 12 }}
+{{- end }}
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector-metrics/deployment.yaml:116]:
```suggestion
          {{- if .Values.otelCollectorMetrics.extraVolumeMounts }}
          {{- toYaml .Values.otelCollectorMetrics.extraVolumeMounts | nindent 12 }}
          {{- end }}
```

add indentation

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml [hunk 2]
```diff
@@ -147,6 +150,9 @@ spec:
         - name: otel-collector-metrics-config-vol
           configMap:
             name: {{ include "otelCollectorMetrics.fullname" . }}
+{{- if .Values.otelCollectorMetrics.extraVolumes }}
+{{ toYaml .Values.otelCollectorMetrics.extraVolumes | indent 8 }}
+{{- end }}
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector-metrics/deployment.yaml:155]: same as above

File: charts/signoz/templates/otel-collector/deployment.yaml [hunk 3]
```diff
@@ -116,6 +116,9 @@ spec:
           volumeMounts:
             - name: otel-collector-config-vol
               mountPath: /conf
+{{- if .Values.otelCollector.extraVolumeMounts }}
+{{ toYaml .Values.otelCollector.extraVolumeMounts | nindent 12 }}
+{{- end }}
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector/deployment.yaml:121]: same as above

File: charts/signoz/templates/otel-collector/deployment.yaml [hunk 4]
```diff
@@ -150,6 +153,9 @@ spec:
         - name: otel-collector-config-vol
           configMap:
             name: {{ include "otelCollector.fullname" . }}
+{{- if .Values.otelCollector.extraVolumes }}
+{{ toYaml .Values.otelCollector.extraVolumes | indent 8 }}
+{{- end }}
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector/deployment.yaml:158]: same as above

**Review by prashant-shahi — CHANGES_REQUESTED** [review]
Looks good, except for the minor indentations. left comments.

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml (hunk 2 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector-metrics/deployment.yaml:155]: The problem with that is if you ident it the output look like this:
```
      volumes:
        - name: otel-collector-metrics-config-vol
          configMap:
            name: test-signoz-otel-collector-metrics
                - name: test-pvc
          persistentVolumeClaim:
            claimName: pvc-xxxxxxxxxx
```
And give this error:
```
Error: YAML parse error on signoz/templates/otel-collector-metrics/deployment.yaml: error converting YAML to JSON: yaml: line 150: mapping values are not allowed in this context
helm.go:84: [debug] error converting YAML to JSON: yaml: line 150: mapping values are not allowed in this context
```
Except you have another solution, no indentation work

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml (hunk 1 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector-metrics/deployment.yaml:116]: Effectivly, it dosn't change anything

File: charts/signoz/templates/otel-collector/deployment.yaml (hunk 3 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector/deployment.yaml:121]: Effectivly, it dosn't change anything

File: charts/signoz/templates/otel-collector/deployment.yaml (hunk 4 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector/deployment.yaml:158]: Same as your second comment

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml (hunk 2 above)
Comment (prashant-shahi) [reply charts/signoz/templates/otel-collector-metrics/deployment.yaml:155]: You could try `nindent` instead.

Can you try the following?

```suggestion
        {{- if .Values.otelCollectorMetrics.extraVolumes }}
          {{- toYaml .Values.otelCollectorMetrics.extraVolumes | nindent 8 }}
        {{- end }}
```

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml [hunk 5]
```diff
@@ -111,6 +111,9 @@ spec:
           volumeMounts:
             - name: otel-collector-metrics-config-vol
               mountPath: /conf
+              {{- if .Values.otelCollectorMetrics.extraVolumeMounts }}
+              {{- toYaml .Values.otelCollectorMetrics.extraVolumeMounts | nindent 12 }}
+              {{- end }}
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector-metrics/deployment.yaml:116]: minor indentation pattern suggestion

```suggestion
            {{- if .Values.otelCollectorMetrics.extraVolumeMounts }}
              {{- toYaml .Values.otelCollectorMetrics.extraVolumeMounts | nindent 12 }}
            {{- end }}
```

File: charts/signoz/templates/otel-collector/deployment.yaml [hunk 6]
```diff
@@ -116,6 +116,9 @@ spec:
           volumeMounts:
             - name: otel-collector-config-vol
               mountPath: /conf
+            {{- if .Values.otelCollector.extraVolumeMounts }}
+            {{ toYaml .Values.otelCollector.extraVolumeMounts | nindent 12 }}
+            {{- end }}
```
Comment (prashant-shahi) [inline charts/signoz/templates/otel-collector/deployment.yaml:121]: minor indentation pattern suggestion

```suggestion
            {{- if .Values.otelCollector.extraVolumeMounts }}
              {{ toYaml .Values.otelCollector.extraVolumeMounts | nindent 12 }}
            {{- end }}
```

**Review by prashant-shahi — CHANGES_REQUESTED** [review]
suggestions and left comments

File: charts/signoz/templates/otel-collector-metrics/deployment.yaml (hunk 2 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector-metrics/deployment.yaml]: Oh that's work ! Thx for that, Done in: https://github.com/SigNoz/charts/pull/270/commits/25dfe02fbfa9bbce3ae37f6e9814224858b2dfcf and https://github.com/SigNoz/charts/pull/270/commits/f131c9ed76a8f77f621640fe13e63edf9d2e909d

File: charts/signoz/templates/otel-collector/deployment.yaml (hunk 4 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector/deployment.yaml]: Done in: https://github.com/SigNoz/charts/commit/25dfe02fbfa9bbce3ae37f6e9814224858b2dfcf and https://github.com/SigNoz/charts/commit/f131c9ed76a8f77f621640fe13e63edf9d2e909d

File: charts/signoz/templates/otel-collector/deployment.yaml (hunk 6 above)
Comment (rchallie) [reply charts/signoz/templates/otel-collector/deployment.yaml:121]: Done in : https://github.com/SigNoz/charts/pull/270/commits/3379cd38b5cdfd002cb6c8a0097cc309320b0f00

Comment (rchallie) [reply charts/signoz/templates/otel-collector/deployment.yaml]: Done in: https://github.com/SigNoz/charts/pull/270/commits/4d9fdd7b27779b00b7245554d0aade527db1bffc

**Review by prashant-shahi — APPROVED** [review]

**Comment by prashant-shahi** [comment]
@rchallie Thanks for the PR contribution. 🎉 
It will be part of the next chart release.

**Pull request merged**